test/simulator/timestamp/src/include/session_simulator.h

```diff
@@ -36,17 +36,22 @@ class session_simulator {
     public:
     session_simulator();
     ~session_simulator() = default;
-    void begin_transaction();
+    int begin_transaction(const std::string &);
     void rollback_transaction();
     void commit_transaction();
     int decode_timestamp_config_map(
       std::map<std::string, std::string> &, uint64_t &, uint64_t &, uint64_t &, uint64_t &);
     int timestamp_transaction(const std::string &);
     int timestamp_transaction_uint(const std::string &, uint64_t);
+    uint64_t get_commit_timestamp();
```

Comment:
Usually, getters are declared as `const` as they don't modify anything in the class:
```suggestion
    uint64_t get_commit_timestamp() const;
```
This applies to all getters and any function that follows that rule. You will need to update the cpp file accordingly.